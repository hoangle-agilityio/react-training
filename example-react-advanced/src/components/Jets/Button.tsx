import { memo } from 'react';

interface ButtonProps {
  buttonName: string;
  className: string;
  onClick?: () => void;
}

const Button = ({ buttonName, className, onClick }: ButtonProps): JSX.Element => {
  return (
    <button data-testid="button" className={className} onClick={onClick}>
      <span className="button-title">{buttonName}</span>
    </button>
  );
};

export default memo(Button);
