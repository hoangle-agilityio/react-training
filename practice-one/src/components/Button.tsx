interface ButtonProps {
  typeButton: "button" | "submit" | "reset";
  size: "sm" | "md"| "lg";
  buttonColor: "light" | "info" | "danger";
  customize?: string;
  label: string;
}

export default function Button({ typeButton, size, buttonColor, customize, label }: ButtonProps) {
  const separateClass = customize ? customize : "";

  return (
    <button
      type={typeButton}
      className={`${separateClass} btn btn-${buttonColor} btn--${size}`}
    >
      {label}
    </button>
  );
}
