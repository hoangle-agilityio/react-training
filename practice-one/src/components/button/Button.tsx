import "./button.scss";

interface ButtonProps {
  typeButton: "button" | "submit" | "reset";
  size?: "sm" | "md"| "lg";
  buttonColor: "light" | "info" | "danger";
  customize?: string;
  label: string;
}

export default function Button({ typeButton, size, buttonColor, customize, label }: ButtonProps) {
  const separateClass: string = customize ? customize : "";
  const buttonSize: string = size ? `btn--${size}` : "";

  return (
    <button
      type={typeButton}
      className={`${separateClass} btn btn-${buttonColor} ${buttonSize}`}
    >
      {label}
    </button>
  );
}
