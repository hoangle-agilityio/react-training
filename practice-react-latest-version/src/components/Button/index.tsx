// Libs
import * as React from "react";
import { memo } from "react";
import { Button as ReactButton, ButtonProps } from "@chakra-ui/react";

interface Props {
  label: React.ReactNode;
  disabled?: boolean;
  type?: "primary" | "secondary" | "text";
  styles?: ButtonProps;
  onClick?: () => void;
}

const Button = ({
  label,
  disabled = false,
  type = "primary",
  styles,
  onClick,
}: Props) => {
  let props = null;
  switch (type) {
    case "primary":
      props = {
        bgColor: "blue.300",
        color: "white.100",
      };
      break;
    case "secondary":
      props = {
        bgColor: "transparent",
        color: "red.200",
      };
      break;
    default:
      props = {
        bgColor: "transparent",
      };
  }

  return (
    <ReactButton
      {...props}
      {...styles}
      fontSize="base"
      fontFamily="medium"
      fontWeight="500"
      _hover={{
        opacity: "70%",
        bgColor: `${type === "secondary" ? "gray.200" : null}`,
      }}
      isDisabled={disabled}
      onClick={onClick}
    >
      {label}
    </ReactButton>
  );
};

export default memo(Button);
