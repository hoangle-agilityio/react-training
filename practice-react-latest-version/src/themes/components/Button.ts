export const Button = {
  baseStyle: {
    fontFamily: "medium",
    fontWeight: "500",
    _hover: {
      opacity: "70%",
    },
  },
  variants: {
    primary: {
      bgColor: "background.primary",
      color: "brand.100",
      fontSize: "base",
    },
    secondary: {
      bgColor: "transparent",
      color: "text.reversal",
      fontSize: "base",
      _hover: {
        bgColor: "background.default",
      },
    },
  },
  defaultProps: {
    variant: "primary",
  },
};
