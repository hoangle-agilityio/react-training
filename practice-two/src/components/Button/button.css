.btn {
  border: none;
  border-radius: 5px;
  margin: 10px;
  cursor: pointer;
}

.button-edit {
  background-color: #037BFF;
  color: #FFFFFF;
  padding: 10px 20px;
  font-size: 14px;
}

.button-delete {
  background-color: #DC3545;
  color: #FFFFFF;
  padding: 10px 20px;
  font-size: 14px;
}
