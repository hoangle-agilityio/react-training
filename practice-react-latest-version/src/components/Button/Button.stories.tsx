// Libs
import * as React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

// Components
import Button from "components/Button";

export default {
  title: "Components/Button",
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  ...Primary.args,
  label: "+ Add customer",
  type: "primary",
};

export const Secondary = Template.bind({});
Secondary.args = {
  label: "Cancel",
  type: "secondary",
};

export const Text = Template.bind({});
Text.args = {
  label: "Edit",
  type: "text",
  styles: {
    color: "blue.200",
  },
};
