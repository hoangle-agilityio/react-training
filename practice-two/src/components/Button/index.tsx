import { memo } from "react";
import "./button.css";

interface ButtonProps {
  buttonName: string;
  type: string;
  onClick?: () => void;
}

function Button({ buttonName, type, onClick }: ButtonProps): JSX.Element {
  return (
    <button className={`btn btn-${type}`} onClick={onClick}>
      {buttonName}
    </button>
  );
}

export default memo(Button);
