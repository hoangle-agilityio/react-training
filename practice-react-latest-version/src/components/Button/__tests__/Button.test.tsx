import { fireEvent, render } from "@testing-library/react";
import * as React from "react";
import Button from "../index";

const props = {
  label: "Shared button",
  onClick: jest.fn(),
};

describe("Button Component", () => {
  test("Should match snapshoot for default Button component", () => {
    const dom = render(<Button {...props} />);

    expect(dom).toMatchSnapshot();
  });

  test("Should display primary button", () => {
    const dom = render(<Button label="Primary Button" type="primary" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should display secondary button", () => {
    const dom = render(<Button label="Secondary Button" type="secondary" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should display button with type is text", () => {
    const dom = render(
      <Button label="Text" type="text" styles={{ color: "blue.200" }} />
    );

    expect(dom).toMatchSnapshot();
  });

  test("Should display disabled button", () => {
    const { getByText } = render(
      <Button {...props} type="primary" disabled={true} />
    );
    const button = getByText("Shared button");

    fireEvent.click(button);
    expect(props.onClick).toHaveBeenCalledTimes(0);
  });

  test("Should handle onClick event", () => {
    const { getByText } = render(<Button {...props} />);
    const button = getByText("Shared button");

    fireEvent.click(button);
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
