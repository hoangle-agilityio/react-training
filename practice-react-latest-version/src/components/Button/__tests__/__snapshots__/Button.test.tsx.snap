// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should display button with type is text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="chakra-button css-b927f8"
        type="button"
      >
        Text
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="chakra-button css-b927f8"
      type="button"
    >
      Text
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button Component Should display primary button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="chakra-button css-3v0amg"
        type="button"
      >
        Primary Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="chakra-button css-3v0amg"
      type="button"
    >
      Primary Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button Component Should display secondary button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="chakra-button css-1ohcnjw"
        type="button"
      >
        Secondary Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="chakra-button css-1ohcnjw"
      type="button"
    >
      Secondary Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button Component Should match snapshoot for default Button component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="chakra-button css-3v0amg"
        type="button"
      >
        Shared button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="chakra-button css-3v0amg"
      type="button"
    >
      Shared button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
